service/history/timerQueueProcessor.go

```diff
@@ -717,6 +722,42 @@ Update_History_Loop:
 	return ErrMaxAttemptsExceeded
 }
 
+func (t *timerQueueProcessorImpl) processWorkflowTimeout(
+	context *workflowExecutionContext, task *persistence.TimerTaskInfo) error {
+	t.metricsClient.IncCounter(metrics.TimerTaskWorkflowTimeoutScope, metrics.TaskRequests)
+	sw := t.metricsClient.StartTimer(metrics.TimerTaskWorkflowTimeoutScope, metrics.TaskLatency)
+	defer sw.Stop()
+
+Update_History_Loop:
+	for attempt := 0; attempt < conditionalRetryCount; attempt++ {
+		msBuilder, err1 := context.loadWorkflowExecution()
+		if err1 != nil {
```

Comment:
We need to check if the workflow completed and that is what cause the load failure.  Otherwise we will keep on retrying.